### PR TITLE
Avoid retally by changing the enddate during recalc

### PR DESCRIPTION
**Pull request by kartikshahc**
Jira issue: [[SWATCH-1780](https://issues.redhat.com/browse/SWATCH-1780)]

## Description

This is a PR same as what we deployed for hotfix branch to stop retally process hence I have put no test steps.

### CODE REVIEW

**Review by kahowell — APPROVED** [review]

**Review by kahowell — CHANGES_REQUESTED** [review]
Needs some test updates

File: src/main/java/org/candlepin/subscriptions/event/EventController.java [hunk 1]
```diff
@@ -78,9 +79,17 @@ public Stream<Event> fetchEventsInTimeRange(
   }
 
   public Stream<Event> fetchEventsInTimeRangeByServiceType(
-      String orgId, String serviceType, OffsetDateTime begin, OffsetDateTime end) {
+      String orgId,
+      String serviceType,
+      OffsetDateTime begin,
+      OffsetDateTime end,
+      OffsetDateTime processEndDateTime) {
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/event/EventController.java:86]: I think `asOfDateTime` might be a better name. You should add some comments around how this is used. Something like: "asOfDateTime is used to ignore events that are recorded outside a given hourly tally's window. Those events will be processed during the next hourly tally, and filtering them out here ensures they won't be processed twice."

**Review by kahowell — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/subscriptions/event/EventController.java (hunk 1 above)
Comment (kartikshahc) [reply src/main/java/org/candlepin/subscriptions/event/EventController.java:86]: sounds good. Will add comments and change the variable name

Comment (kartikshahc) [reply src/main/java/org/candlepin/subscriptions/event/EventController.java]: @kahowell have added the comment and changed the variable name.

**Review by kahowell — APPROVED** [review]

**Review by ssidhaye — APPROVED** [review]
Tested changes in EE for past events the behavior is as expected.

**Pull request merged**